.github/workflows/docker.yml

```diff
@@ -0,0 +1,63 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+name: Build and Push PDO Docker Images
+
+on:
+  workflow_dispatch:
+
+  pull_request:
+    types: [closed]
+    branches: [main]
+
+jobs:
+
+  docker_build:
+
+    if: >
+      github.event.name == 'workflow_dispatch' ||
+      github.event.name == 'pull_request' && github.event.pull_request.merged == true
```

Comment:
Right now we would generate the dockers for every PR, whether it is a version upgrade or not.  But i guess there is no easy way to test whether this is a version upgrade PR?   That said, as long as storage is not an issue, i guess there is loittle cost for doing it on all PRs (the actual build should be "free" if github caching works properly as the CI already should have built them?)